src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -818,8 +818,7 @@ public boolean equals(Object obj) {
                          && p.isInteresting(changeBasedEvent.getChange().getProject(),
                                             changeBasedEvent.getChange().getBranch(),
                                             changeBasedEvent.getChange().getTopic())) {
-                        if (isFileTriggerEnabled() && p.getFilePaths() != null
-                                && p.getFilePaths().size() > 0) {
+                        if (isFileTriggerEnabled()) {
```

Comment:
This scares me a little, because it means that there will be a callback to query files for each change-based event that enters Jenkins instead of if there is interest in files. (Indicated by the extra ssh commands you needed to mock in the tests).
I think you can get the same effect if you just also check ```p.getForbiddenFilePaths()```.